[BE][DTensor] merge random init test to test_random_ops.py

XilunWu opened the pull request:
Random Ops behavior becomes different on CUDA device and others in PR https://github.com/pytorch/pytorch/pull/98199 . As a result, test of DTensor random initialization in test_init.py was skipped. This PR fixes it by having different testing assertion for different types of device and also merges existing duplicated test.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99014
* __->__ #98874

XilunWu committed:
[BE][DTensor] merge random init test to test_random_ops.py

[ghstack-poisoned]

XilunWu commented:
@pytorchbot rebase

XilunWu committed:
Update on "[BE][DTensor] merge random init test to test_random_ops.py"

Random Ops behavior becomes different on CUDA device and others in PR https://github.com/pytorch/pytorch/pull/98199 . As a result, test of DTensor random initialization in test_init.py was skipped. This PR fixes it by having different testing assertion for different types of device and also merges existing duplicated test.




[ghstack-poisoned]

wanchaol commented on test/distributed/_tensor/test_random_ops.py:
```diff
@@ -26,6 +27,63 @@
+                    slice_idx = [
+                        slice(input_size[0]),
+                        slice(
+                            other_rank * input_size[1], (other_rank + 1) * input_size[1]
```
what if `other_rank` is the last rank, would it still generate the right shape?

XilunWu replied:
Yes. I tried the following example:
```
>>> import torch
>>> tensor = torch.arange(32).reshape(4, 8)
>>> last_rank = 3
>>> slice_idx = [slice(4), slice(last_rank * 2, (last_rank + 1) * 2)]
>>> tensor[slice_idx]
tensor([[ 6,  7],
        [14, 15],
        [22, 23],
        [30, 31]])
```

wanchaol reviewed (approved):
looks good! Only have one question

XilunWu commented:
@pytorchmergebot merge

XilunWu committed:
Update on "[BE][DTensor] merge random init test to test_random_ops.py"

Random Ops behavior becomes different on CUDA device and others in PR https://github.com/pytorch/pytorch/pull/98199 . As a result, test of DTensor random initialization in test_init.py was skipped. This PR fixes it by having different testing assertion for different types of device and also merges existing duplicated test.




[ghstack-poisoned]

XilunWu commented:
@pytorchmergebot merge

malfet commented:
https://github.com/pytorch/pytorch/commit/49cd650e2bd6d84ee936adac62c379bc8b67dd53 has been merged to trunk, so closing
